### PR TITLE
CIP-0989? | ISPO KYC_CDD

### PR DESCRIPTION
**Pull request by johnalanwoods**
We introduce a mechanism to enable delegators to securely provide identifying information about themselves, which can be leveraged by SPOs to perform appropriate KYC/CDD. The approach also provides bilateral privacy between participants.

This article outlines a technical approach to ensure SPO (stake pool operators) are empowered to receive data regarding the identity of ADA holders delegating their stake to a pool. 

John Woods
Director of Architecture for Cardano
IOG

### CODE REVIEW

**Comment by KtorZ** [comment]
@johnalanwoods we're wondering about the motivation behind the proposal, in particular from the SPO side? We're thinking this is something worth maybe presenting in the next discord SPO call to get attention from relevant parties? 

Also, we'd invite you to share it on the Cardano forum: https://forum.cardano.org/c/developers/cips/122 where CIPs gets discussed with a broader community (including some SPOs)

**Comment by rphair** [comment]
@johnalanwoods @KtorZ also I've asked SPOCRA to chime in about this 😎 https://members.spocra.io/posts/22505688

**Comment by johnalanwoods** [comment]
Sounds good both.

Motivation is to get ahead of things. Being proactive about providing a mechanism rather than reactive to any possible need to the future.

File: CIP-0989/README.md [hunk 1]
```diff
@@ -0,0 +1,131 @@
...
+
+### DID Privacy
+In order to submit a valid DID for a given ISPO event, an end-user must include a DID in the transaction metadata which is being submitted to register stake.
+This DID information should be confidential and only visible to the concerned parties (end-user and SPO). Thus we propose a mechanism for cryptographic key agreement between end-user and SPO in a non-interaction fashion, the resulting key material can be used to protect the DID data.
+
+We will leverage `static:static ECDH` (Elliptic curve Diffie-Hellman) in order to agree a symmetric key, this process will leverage the end-user's existing staking key pair, and require a **new hot key pair** on the SPO side, known as the *DID decryption key pair*. This process assumes the same curve and curve generator point are being used by both parties.
+
+
+#### Key Agreement Process
+$$
```
Comment (SebastienGllmt) [inline CIP-0989/README.md:98]: This won't render properly on markdown.

**Comment by rphair** [comment]
> Sounds good both.

@johnalanwoods at the CIP Editors' meeting yesterday we were talking about this one again & still no feedback from SPOs.  I wrote to Rebecca Hopwood @ IOG and she says you're a regular on the SPO calls... did you ever share this there?  If not, could you poll the SPOs at the next call to feed back here?

I would also like to see this discussed on the Cardano Forum as linked above, though I haven't seen much forum participation from the leads of the big token projects.  Yet some potential CIPs there e.g. those having to do with PoS mechanics have had a lot of detailed discussion there so I still think it's worth a try.

It may be most helpful to ping the project leads of ISPOs of the last year, and the ones active now, for feedback.  Also World Mobile Token didn't do an ISPO but they used KYC for their ICO, storing the validations as metadata somehow.  If you need help finding the project leads or tagging them on Github let me know & I'll do what I can.

**Comment by link78** [comment]
Hello John, thanks for submitting this CIP, when i was reading this cip, I came to one phrase: This article outlines a technical approach to ensure SPO (stake pool operators) are empowered to receive data regarding the identity of ADA holders delegating their stake to a pool." SPO will receive data regarding the identity of the ADA holder. Most ada holder want to stay anonymous. What will happen if the SPO refuse to collect user information?

**Comment by RaceSpeed** [comment]
Credentials and Decentralized Id

Tests 
[](https://github.com/Catalyst-Circle/Catalyst-Circle-Admin-Coordination/issues/49)
[](https://github.com/Catalyst-Circle/Catalyst-Circle-Admin-Coordination/issues/25)

WALLET creation -Education- Tooling- 
-Data input-
-Email-
-Stake Key-
-Matrix encryption verifications
Gaining Identity- through proof of life/work

Shielding of sensitive information to improve utility and building Reputation 
F8 Proposal:
https://cardano.ideascale.com/c/idea/401391

Collect 7 points of Verification- Email- Name(username)- (Any socials or profiles/resume) (Stake1) (Region)(Matrix ID) (Crafts NFT) Cost- 2ADA

Collect 7 credentials from Peers. (NFT’s or Fungible token trades) Goal: (collect 10 ADA) to Stake Stakekey +( Matrix name) (collect 1.3-2.2ADA) 7 proof’s of verification to reach Staking- Add key by sending credentials and new inputs. 

-Craft own fungible token to give to others. 
-Give 7 credentials to peers for rewards. ( give 7 fungible or NFT proof of life interactions) -14 ADA

Use verified ID for proof of life and work in ecosystem and Attach to DID profile
Verified by peers- New User Contributes and delegates to community (POOL) after ID points reached. 

Example of token - b3f5cdbf8d987a1d04b72a2b728e748a5d58e9d0f5e439b0890cc212
image
[](https://pool.pm/asset1zeh0fq60yl2kyct4vtz6pwmrz45cj0lgryskhv)

Other tests 
Transparency 
xxxxx did not have ANY NFT’s in his wallet
addr1qyc65d00lprtyyjl47ew88mftta9tvww0x5tnxmldfq0mww66grw5asm7gja49xx44ht3jvmlqcqnta5fwv67xhu72kq3jyynv
xxxx) :
addr1q9dqn650hrwy2dyeda3uzuqe5s4ea6xpqtlkeu6uzhngx5kmfy3lzvgsttrudre8nust430tpxgkc27mzrqwdcx76sss2qp0f9
xxxxxt
addr1qxzftwvmsnfawl4ytxf9n6hn25xwru5uszutrmsyzcz64u0s46qm0tqw7m2mmz4m05d4wnu4cvayw8g42wmesww0a30su8hvl3
xxxx did no have ANY NFT’s in his wallet
addr1qypr5t6prh924z58h9ga8q5w842njth4uqth5zn0k7dllr5rdcnj6lwu3zw854y76snq07kauuw4uj3s3hz2v6k4t5gquxd83r
xxxxneeded NFT’s in their wallet
addr1qxrha8nx8qj4j8dqn5jr2ju4pmpc2wra85fd95dkpnaw9rndc0du2ykudjxmwc8rsfhky0ppm42h0twk22ucx7qyswtqr6536d
SOLVE
3 month challenge- HOW many nft’s can they collect in 3 months
Return on Intention- The intention is the put NFT’s in each person’s wallet and show Return and Proof
Document and track what is going on in the challenge- ADVISE and provide guidance- Challenge to help without making any decision. 3 month cycle and review

SENT xxxxxx
Proof-
https://explorer.cardano.org/en/transaction?id=7d5490299aee5915b2c648f6f7947df71f1c708738eaca129dda21be21036815
https://explorer.cardano.org/en/transaction?id=bf08aeadca7e93ee3ed2826b7dc735e58f4e36fbe53d35be5514d8ae7fb5a1ea
https://explorer.cardano.org/en/transaction?id=2037d5a5ad15a13832adf34043605913969e59ddaf194b5e633ec80c1071a827
https://explorer.cardano.org/en/transaction?id=18f7fb69a427cf66b11823f3331839ec2ad1dacd6256bd704add57cda485c1c2
https://explorer.cardano.org/en/transaction?id=f7d87ea3ef32c4e69eea57043b5aa96c81892b80da2ac1ae12120fad36da9091

To check each wallet- Please check here
Fxx
https://pool.pm/addr1qyc65d00lprtyyjl47ew88mftta9tvww0x5tnxmldfq0mww66grw5asm7gja49xx44ht3jvmlqcqnta5fwv67xhu72kq3jyynv

Jxxx
https://pool.pm/addr1q9dqn650hrwy2dyeda3uzuqe5s4ea6xpqtlkeu6uzhngx5kmfy3lzvgsttrudre8nust430tpxgkc27mzrqwdcx76sss2qp0f9

Nxxx
https://pool.pm/addr1qxzftwvmsnfawl4ytxf9n6hn25xwru5uszutrmsyzcz64u0s46qm0tqw7m2mmz4m05d4wnu4cvayw8g42wmesww0a30su8hvl3

Axxx
https://pool.pm/addr1qypr5t6prh924z58h9ga8q5w842njth4uqth5zn0k7dllr5rdcnj6lwu3zw854y76snq07kauuw4uj3s3hz2v6k4t5gquxd83r

Nxxx
https://pool.pm/addr1qxrha8nx8qj4j8dqn5jr2ju4pmpc2wra85fd95dkpnaw9rndc0du2ykudjxmwc8rsfhky0ppm42h0twk22ucx7qyswtqr6536d
Conclusion of Part 1 Return on intent
Total assets in ADA XXX Plus
2.317899 + 2.238285 + 2.23285 + 2.49837 +2.289489
11.576893 Sum total for Transactions.
Investment into NFT’s 
Asset id’s – approx. 3 ADA mint for each or accepted by trade.
436865717565426f78
41
4152474f30
4b494e47424954
4a415641
All tasks have been completed and follow up will be in 3 months on progress.

**Comment by rphair** [comment]
> Credentials and Decentralized Id

@RaceSpeed was your [last comment](https://github.com/cardano-foundation/CIPs/pull/241#issuecomment-1137778337) intended for this PR and not some Catalyst Circle discussion?

If so I do think that NFT based means of identity would be relevant to ISPOs and KYC in general, though it would help if you'd explain how that relates to the method of DID storage that's being proposed here.

In either case perhaps you could edit down your last comment so we can focus on whatever comment(s) you're making that relates to the text of this proposal. 😎

File: CIP-0989/README.md (hunk 1 above)
Comment (rphair) [reply CIP-0989/README.md:98]: I tried to make this work on GitHub with a simple change to the TeX formatted block & couldn't.  @johnalanwoods the math terms just need to be broken out from the text with single `$` delimiters.

**Comment by johnalanwoods** [comment]
> > Sounds good both.
> 
> @johnalanwoods at the CIP Editors' meeting yesterday we were talking about this one again & still no feedback from SPOs. I wrote to Rebecca Hopwood @ IOG and she says you're a regular on the SPO calls... did you ever share this there? If not, could you poll the SPOs at the next call to feed back here?
> 
> I would also like to see this discussed on the Cardano Forum as @KtorZ suggested, though I haven't seen much forum participation from the leads of the big token projects. Yet some potential CIPs there e.g. those having to do with PoS mechanics have had a lot of detailed discussion there so I still think it's worth a try.
> 
> It may be most helpful to ping the project leads of ISPOs of the last year, and the ones active now, for feedback. Also World Mobile Token didn't do an ISPO but they used KYC for their ICO, storing the validations as metadata somehow. If you need help finding the project leads or tagging them on Github let me know & I'll do what I can.

Yep - I think we need to discuss this on the SPO call.

**Comment by johnalanwoods** [comment]
> > empowered to receive data regarding the identity of ADA holders delegating their stake to a pool.
> 
> @johnalanwoods there may be others who read this besides @link78 who get the wrong idea from the use of the word "empowered" ... though it was clear enough to me that this framework would be there _only_ if it suits certain SPOs running an ISPO, rather than imposed on _all_ stake pools & delegators. Maybe some different language can be chosen that's less likely to make people think it's obligatory?
> 
> > _enabled_ to receive data ...
> 
> or something more explicit:
> 
> > _enabled_, if it suits the conditions of a stake pool's ISPO, to receive data ...

You're absolutely right. This is _completely_ optional. It's only applicable in certain jurisdictions, where specific laws apply.

This does *not* apply to regular delegation.

**Comment by johnalanwoods** [comment]
> Hello John, thanks for submitting this CIP, when i was reading this cip, I came to one phrase: This article outlines a technical approach to ensure SPO (stake pool operators) are empowered to receive data regarding the identity of ADA holders delegating their stake to a pool." SPO will receive data regarding the identity of the ADA holder. Most ada holder want to stay anonymous. What will happen if the SPO refuse to collect user information?

Nothing. This is optional. It's really only to be used when an SPO is issuing a token, and even then, only if the SPO wishes to comply with regulation.

**Comment by rphair** [comment]
> Yep - I think we need to discuss this on the SPO call.

Yes @johnalanwoods  I think that will be needed, since as requested they put something in today's monthly newsletter: https://mailchi.mp/iohk/spo-digest-february-675438?e=d55ecc2aed
... but it's little more than a footnote and wasn't highlighted either for administrators of ISPO projects or SPO's interested in the KYC or DID's:
> Be sure to check out [CIP-0989](https://iohk.us20.list-manage.com/track/click?u=26d3b656ecc43aa6f3063eaed&id=74867caf92&e=d55ecc2aed) which may be of interest to the SPO community.

**Comment by rphair** [comment]
also I had a message from another marketing team member this morning saying there was a Twitter Space today on the subject (I was invited but missed it).  @johnalanwoods if there was any feedback from community members joining that session could you relay it here? - https://twitter.com/NodeSanto/status/1529889457473826824

Also I was thinking if this Twitter Space was announced somewhere it must have been the Discord server (it wasn't on Telegram or email), so there may also be come chat in the relevant IOG Discord channels about the proposal.

**Comment by rphair** [comment]
@johnalanwoods FYI a new user on the Cardano Forum tried to open up a thread on this topic, but it was removed; probably because it was a link and nothing else.  Since therefore it seems there's some need to discuss this on the forum I've created a topic here: https://forum.cardano.org/t/ispo-kyc-proposal-cip-0989/102084

**Comment by rphair** [comment]
call for discussion today in monthly SPO Digest here: https://mailchi.mp/iohk/spo-digest-february-675602

**Comment by rphair** [comment]
@johnalanwoods given [this announcement](https://twitter.com/JohnAlanWoods/status/1545060330467799040) (with best of luck to you) will you be advocating this PR further, or are you leaving it up to the editors & community?

**Comment by johnalanwoods** [comment]
Hi happy to contribute on this still. It makes sense that I do not own it, but still happy to work with others.

Thanks

**Review by rphair — APPROVED** [review]
I think this represents the best achievement that current stakeholders & the community have been able to produce in the relevant area.  Perhaps this is because ISPOs aren't so abundant anymore, this is also a less visible issue than it used to be... but will be nice to have a framework in place that new ISPOs could use to get started quickly if they need to do this for compliance reasons.

After several calls this year to find fault or recommend alternative means of doing KYC for stake pools, we don't have any substantial revisions to this proposal: only some suggestions about the language so nobody thinks KYC will be obligatory for stake pools (something most people will know by common sense).

So I think we are throwing a way a good bit of equity if we don't merge this, considering @johnalanwoods has already produced something workable and shouldn't be called back to advocate for it.  Having seen this through a couple of CIP meetings already, as long as the header fields are OK with @KtorZ then I think this can be merged anytime.

**Comment by KtorZ** [comment]
Closing this for inactivity and for the sake of cleaning up things. The debate / feature can be discussed anew be re-opening the PR or making a new one championed by a new author.

**Pull request closed without merging**